services/wallet/service.go

```diff
@@ -446,11 +447,42 @@ func (service *Service) LinkZebPayWallet(ctx context.Context, walletID uuid.UUID
 		return handlers.WrapError(appctx.ErrNotInContext, msg, http.StatusBadRequest)
 	}
 
+	// validate algorithm used
+	for i := range tok.Headers {
```

Comment:
Is it possible that the header length can be 0, if so then do we want to return here rather then continue processing the request?